embeddings/task/lightning_task/lightning_task.py

```diff
@@ -24,20 +26,43 @@ def __init__(
         self,
         output_path: T_path,
         task_train_kwargs: Dict[str, Any],
+        early_stopping_kwargs: Dict[str, Any],
     ):
         super().__init__()
         self.output_path: Path = Path(output_path)
         self.task_train_kwargs = task_train_kwargs
+        self.early_stopping_kwargs = early_stopping_kwargs
         self.model: Optional[HuggingFaceLightningModule] = None
         self.trainer: Optional[pl.Trainer] = None
 
+    @property
+    def best_epoch(self) -> Optional[float]:
+        if self.trainer is None:
+            return
+
+        for callback in self.trainer.callbacks:
+            if isinstance(callback, BestEpochCallback):
+                return callback.best_epoch
+
+    @property
+    def best_validation_score(self) -> Optional[float]:
+        if self.trainer is None:
+            return
+
+        for callback in self.trainer.callbacks:
+            if isinstance(callback, BestEpochCallback):
+                return callback.best_score
+
     def fit(
         self,
         data: HuggingFaceDataModule,
     ) -> None:
         if not self.model:
             raise self.MODEL_UNDEFINED_EXCEPTION
-        self.trainer = pl.Trainer(default_root_dir=str(self.output_path), **self.task_train_kwargs)
+        self.trainer = pl.Trainer(
+            default_root_dir=str(self.output_path),
+            callbacks=[BestEpochCallback(), EarlyStopping(**self.early_stopping_kwargs)],
```

Comment:
It seems that we are enforcing the usage of these callbacks on the user, is it intended?